# Twitter Sentiment Analysis: LuxWorld and Travel to Earn

Currently, sentiment analysis is one of the most popular research techniques to understand the opinions and feelings of the people based on the text data. Machine learning is also one of the popular tools for analyzing the data based on the large amount of data. Nowadays, many researchers try to use the good points of the machine learning to perform the sentiment and opinion mining. 

The present study analyzes the sentiments of Voice of the Customer from LuxWorld and Travel-to-earn users using machine learning algorithms, which learn more quickly and with less complexity than deep learning algorithms. Voice of the Customer is concern about what individual customer is saying about products or services. It means analyzing the reviews and feedback of the customers. VOC is a key element of Customer Experience Management. VOC helps in identifying new opportunities for product inventions.

In this notebook, machine learning algorithms are optimized with English-language tweets expressing sentiments about events, feelings and experiences from October to December 2022.

## Table of Contents
1. Data Scraping and Collection
2. Data Preprocessing
3. Sentiment Analysis
4. EDA


## 1. Data Scraping

Twitter is one of the most important data sources for data scientists. Data scientists can extract data from Twitter with web scraping tools. There are several libraries to do this, such as Tweepy and Snscrape. Tweepy is an easy-to-use Python library but it is limited with extracted tweets. Therefore, we will use Snscrape, which can scrape data from other services like Facebook, Instagram, Reddit, and Telegram.

In [1]:
!pip install snscrape

So, I’m just going to scrape two days of tweets and set the tweet limit to 3000 tweets.

In [2]:
# Importing necessary libraries
import snscrape.modules.twitter as sntwitter
import pandas as pd

In [3]:
def tweets(n_tweets, search_term, language, start_date, end_date):

    # Creating list to append tweet data to
    tweets_list2 = []

    # Using TwitterSearchScraper to scrape data and append tweets to list
    for i,tweet in enumerate(sntwitter.TwitterSearchScraper(f'{search_term} lang:{language} since:{start_date} until:{end_date}').get_items()):
        if i>n_tweets:
            break
        tweets_list2.append([tweet.date, tweet.user.username, tweet.id, tweet.content])

    # Creating a dataframe from the tweets list above
    tweets_df2 = pd.DataFrame(tweets_list2, columns=['Datetime', 'User', 'Tweet Id', 'Text'])
    return tweets_df2

In [4]:
#tweets_df=tweets(100000, 'Luxworld','en', '2022-10-01', '2022-12-26') 
#comment it out once you have gathered your data to avoid running out of the wait time gather the data

In [5]:
#save the dataframe to csv
tweets_df.to_csv("tweets_luxworld.csv", index=False, header=True)

In [6]:
tweets_df.head()

Datetime            User             Tweet Id  \
0 2022-12-25 15:08:39+00:00   0xiqbalSCP001  1607030569472528386   
1 2022-12-25 12:47:32+00:00    PealsSamyeli  1606995057843789828   
2 2022-12-25 11:02:16+00:00   TommyWang2211  1606968564254404610   
3 2022-12-25 10:55:43+00:00  james19_oliver  1606966915850989570   
4 2022-12-25 10:54:06+00:00  james19_oliver  1606966510962216960   

                                                Text  
0  I just won a raffle for @buster_ranger from @L...  
1  @AycanEverley @frjsr714810 @Gameverse_ @ethere...  
2  @LuxWorld_ When #LuxWorld combines #Move2Earn ...  
3  @LucasMaxxx11 @LuxWorld_ maybe both in real li...  
4  @LuxWorld_ LFG #LuxWorld, I believe in your pr...

In [8]:
tweets_df.shape

(6838, 4)

We only collect 6838 tweets of Luxworld, lower than I expected (:D). But it is ok with a new project like LuxWorld. Now we move to collect data for travel to earn trend. 

In [9]:
#tweets_df1=tweets(10000, 'Travel2Earn','en', '2022-10-01', '2022-12-26')

In [10]:
#save the dataframe to csv
tweets_df1.to_csv("tweets_t2e.csv", index=False, header=True)

In [11]:
#check the shape 
tweets_df1.shape

(181, 4)

In [12]:
tweets_df1.head()

Datetime            User             Tweet Id  \
0 2022-12-25 11:02:16+00:00   TommyWang2211  1606968564254404610   
1 2022-12-23 19:30:56+00:00  NobleMoody_357  1606371799872438272   
2 2022-12-21 18:18:57+00:00    KnShayeMedia  1605628909902368769   
3 2022-12-20 16:02:15+00:00    sandy_carter  1605232117494988802   
4 2022-12-20 02:00:01+00:00    GetFitMining  1605020162385534976   

                                                Text  
0  @LuxWorld_ When #LuxWorld combines #Move2Earn ...  
1  #GetFit #Move2Earn #Play2Earn #Travel2Earn rew...  
2  Let’s Get It! #GetFit #Tagcoin #Tagprotocol #G...  
3  And another amazing @unstoppablewebpartner - @...  
4  1⃣ Follow us\n2⃣ Retweet\n3⃣ Tag 3 friends &am...

The amount of tweet for Travel2Earn is quite little so I decide to scrape more data from the trend Move2Earn because Travel2Earn was born by Move2Earn trend. 

In [15]:
#tweets_df2=tweets(10000, 'Move2Earn','en', '2022-10-01', '2022-12-26')

In [16]:
#save the dataframe to csv
tweets_df2.to_csv("tweets_m2e.csv", index=False, header=True)

In [17]:
tweets_df2.head()

Datetime           User             Tweet Id  \
0 2022-12-25 23:51:50+00:00       s13yukio  1607162231590129665   
1 2022-12-25 23:18:16+00:00       haydn239  1607153784676810753   
2 2022-12-25 23:13:38+00:00  CoinHuntWorld  1607152619583062016   
3 2022-12-25 23:11:48+00:00    karadasakin  1607152156204847106   
4 2022-12-25 22:21:19+00:00     lehomonyet  1607139451964395520   

                                                Text  
0  I'm joining TipTop's 1st ever community giveaw...  
1  @ktrainUSA_STEPN Jim and bit boy critiquing SO...  
2  Congratulations @spacecubie, for being the fir...  
3  I'm joining TipTop's 1st ever community giveaw...  
4  Hey @1 @2 - Check out FightOut - the best Move...

In [18]:
tweets_df2.shape

(7810, 4)

So, we're done with scraping data for this notebook. Let's move to perform a Twitter sentiment analysis. 